DeusEx/Classes/DXREnemies.uc

```diff
@@ -156,23 +246,9 @@ function RandomizeSP(ScriptedPawn p, int percent)
     if( IsHuman(p) == False ) return; // only give random weapons to humans
 
     r = initchance();
-    if( chance(11, r) ) wclass = class'WeaponPistol';
-    if( chance(11, r) ) wclass = class'WeaponAssaultGun';
-    if( chance(11, r) ) wclass = class'WeaponMiniCrossbow';
-    if( chance(4, r) ) wclass = class'WeaponGEPGun';
-    if( chance(5, r) ) wclass = class'WeaponAssaultShotgun';
-    if( chance(5, r) ) wclass = class'WeaponEMPGrenade';
-    if( chance(4, r) ) wclass = class'WeaponFlamethrower';
-    if( chance(5, r) ) wclass = class'WeaponGasGrenade';
-    if( chance(5, r) ) wclass = class'WeaponHideAGun';
-    if( chance(5, r) ) wclass = class'WeaponLAM';
-    if( chance(4, r) ) wclass = class'WeaponLAW';
-    if( chance(5, r) ) wclass = class'WeaponNanoVirusGrenade';
-    if( chance(5, r) ) wclass = class'WeaponPepperGun';
-    if( chance(5, r) ) wclass = class'WeaponPlasmaRifle';
-    if( chance(5, r) ) wclass = class'WeaponRifle';
-    if( chance(5, r) ) wclass = class'WeaponSawedOffShotgun';
-    if( chance(5, r) ) wclass = class'WeaponShuriken';
+    for(i=0; i < ArrayCount(randomweapons); i++ ) {
+        if( chance( randomweapons[i].chance, r ) ) wclass = randomweapons[i].type;
+    }
```

Comment:
note this doesn't remove any weapons they may have already had, I changed my config to give a LAW 100% of the time, but at close range the terrorists preferred their pistols until I ran farther away, that's pretty smart!